pkgs/firehose/lib/src/github.dart

```diff
@@ -0,0 +1,182 @@
+  bool get inGithubContext => Platform.environment['GITHUB_ACTIONS'] != null;
+
+  /// The short ref name of the branch or tag that triggered the workflow run.
+  /// This value matches the branch or tag name shown on GitHub. For example,
+  /// `feature-branch-1`.
+  String? get refName => Platform.environment['GITHUB_REF_NAME'];
+
+  http.Client get httpClient => _httpClient ??= http.Client();
+
+  Future<String> callRestApiGet(Uri uri) async {
+    var token = _githubAuthToken!;
+
+    return httpClient.get(uri, headers: {
+      'Authorization': 'Bearer $token',
+      'Accept': 'application/vnd.github+json',
+    }).then((response) {
+      return response.statusCode != 200
+          ? throw RpcException(response.reasonPhrase!)
+          : response.body;
+    });
+  }
+
+  Future<String> callRestApiPost(Uri uri, String body) async {
+    var token = _githubAuthToken!;
+
+    return httpClient
+        .post(uri,
+            headers: {
+              'Authorization': 'Bearer $token',
+              'Accept': 'application/vnd.github+json',
+            },
+            body: body)
+        .then((response) {
+      return response.statusCode != 201
+          ? throw RpcException(response.reasonPhrase!)
+          : response.body;
+    });
+  }
+
+  Future<String> callRestApiPatch(Uri uri, String body) async {
+    var token = _githubAuthToken!;
+
+    return httpClient
+        .patch(uri,
+            headers: {
+              'Authorization': 'Bearer $token',
+              'Accept': 'application/vnd.github+json',
+            },
+            body: body)
+        .then((response) {
+      return response.statusCode != 200
+          ? throw RpcException(response.reasonPhrase!)
+          : response.body;
+    });
+  }
+
+  Future<void> callRestApiDelete(Uri uri) async {
+    var token = _githubAuthToken!;
+
+    return httpClient.delete(uri, headers: {
+      'Authorization': 'Bearer $token',
+      'Accept': 'application/vnd.github+json',
+    }).then((response) {
+      if (response.statusCode != 204) {
+        throw RpcException(response.reasonPhrase!);
+      }
+    });
+  }
+
+  /// Create a new comment on the given PR.
+  Future<String> createComment(
+      String repoSlug, String issueNumber, String commentText) async {
+    var org = repoSlug.split('/')[0];
+    var repo = repoSlug.split('/')[1];
+
+    var result = await callRestApiPost(
+      Uri.parse(
+          'https://api.github.com/repos/$org/$repo/issues/$issueNumber/comments'),
+      jsonEncode({'body': commentText}),
+    );
+    var json = jsonDecode(result) as Map;
+    return json['url'] as String;
+  }
+
+  /// Find a comment on the PR matching the given criteria ([user],
+  /// [searchTerm]). Return the issue ID if a matching comment is found or null
+  /// if there's no match.
+  Future<int?> findCommentId(
+    String repoSlug,
+    String issueNumber, {
+    required String user,
+    String? searchTerm,
+  }) async {
+    var org = repoSlug.split('/')[0];
```

Comment:
Also just an opinion, but would it be cleaner to create getters for `org` and `repo` in the `Github` class in addition to the `repoSlug`?

Reply:
Ah, yeah, I could do that (or make a structured `RepoSlug` class).